### PR TITLE
[wasi] Add build support for using `wasm-opt`

### PR DESCRIPTION
**Pull request by radical**
This adds the build support but does not enable it by default which can be enabled with `$(WasmRunWasmOpt)=true`.
Also, a new `wasm-props.json` file is added to the `wasi` runtime pack. For `browser` this is `emcc-props.json` with a new name.

- [wasm/wasi] Add new wasm-props.json to the runtime pack
   For browser this is `emcc-props.json` renamed. And for wasi, it's a new
   file.
- [wasm/wasi] Extract target for running wasm-opt, and use for both browser, and wasi

### CODE REVIEW

File: src/mono/wasm/build/BrowserWasmApp.targets [hunk 1]
```diff
@@ -205,18 +205,12 @@
       <_EmscriptenPrependPATHTrimmed Include="$([MSBuild]::ValueOrDefault('%(EmscriptenPrependPATH.Identity)\', '').TrimEnd('\/'))" />
       <EmscriptenPrependPATH Remove="@(EmscriptenPrependPATH)" />
       <EmscriptenPrependPATH Include="@(_EmscriptenPrependPATHTrimmed)" />
+
```
Comment (radical) [inline src/mono/wasm/build/BrowserWasmApp.targets:208]:
```suggestion
```

**Comment by radical** [comment]
TODO: wasm-opt run should respect incremental builds

**Comment by radical** [comment]
TODO:
- [ ] check for any regression in file sizes with browser, and blazor

**Review by maraf — APPROVED** [review]
Looks good to me 👍

File: src/mono/wasi/build/WasiApp.targets [hunk 2]
```diff
@@ -370,9 +370,7 @@
     <Error Condition="'$(_WasiOutputFileName)' == ''" Text="Could not determine $(_WasiOutputFileName)" />
   </Target>
 
-  <Target Name="_CheckToolchainIsExpectedVersion">
```
[the remark this answers is not in the document]
Comment (radical) [reply src/mono/wasi/build/WasiApp.targets:373]: I want to implement this in a future PR.

**Comment by radical** [comment]
The hybrid globalization failures are not specific to this PR, and filed as https://github.com/dotnet/runtime/issues/95983 and https://github.com/dotnet/runtime/issues/95981 .

**Pull request merged**